conan/cli/command.py

```diff
@@ -63,10 +66,13 @@ def _help_formatters(self):
         return [formatter for formatter in self._formatters if formatter != "text"]
 
     def _init_formatters(self, parser):
-        formatters = self._help_formatters
-        if formatters:
-            help_message = "Select the output format: {}".format(", ".join(formatters))
+        if self._help_formatters:
+            help_message = "Select the output format: {}".format(", ".join(self._help_formatters))
             parser.add_argument('-f', '--format', action=OnceArgument, help=help_message)
+        # Add the out-file argument if any formatters exist, even the default one not shown in help
+        if self._formatters:
```

Comment:
But this is always ``True``, as it is hardcoded to be ``self._formatters = {"text": lambda x: None}`` and it cannot be removed.
It seems the gap is not being able to differentiate when a user really defines the ``formatters={"text": myformatter`` and that default formatter that does nothing, which is not very clear the intention.